tensorboard/plugins/graph/tf_graph_loader/tf-graph-loader.ts

```diff
@@ -0,0 +1,275 @@
+            'TensorFlow could not trace any graph. Please refer to ' +
+            'https://github.com/tensorflow/tensorboard/issues/1961 for more ' +
+            'information.');
+      }
+
+      // Build the flat graph (consists only of Op nodes).
+
+      // This is the whitelist of inputs on op types that are considered
+      // reference edges. "Assign 0" indicates that the first input to
+      // an OpNode with operation type "Assign" is a reference edge.
+      var refEdges = {};
+      refEdges["Assign 0"] = true;
+      refEdges["AssignAdd 0"] = true;
+      refEdges["AssignSub 0"] = true;
+      refEdges["assign 0"] = true;
+      refEdges["assign_add 0"] = true;
+      refEdges["assign_sub 0"] = true;
+      refEdges["count_up_to 0"] = true;
+      refEdges["ScatterAdd 0"] = true;
+      refEdges["ScatterSub 0"] = true;
+      refEdges["ScatterUpdate 0"] = true;
+      refEdges["scatter_add 0"] = true;
+      refEdges["scatter_sub 0"] = true;
+      refEdges["scatter_update 0"] = true;
+      var buildParams = {
+        enableEmbedding: true,
+        inEmbeddingTypes: ['Const'],
+        outEmbeddingTypes: ['^[a-zA-Z]+Summary$'],
+        refEdges: refEdges
+      };
+      var graphTracker = tf.graph.util.getSubtaskTracker(tracker, 20, 'Graph');
+      return tf.graph.build(graph, buildParams, graphTracker);
+    }, () => {
+      throw new Error('Malformed GraphDef. This can sometimes be caused by a ' +
+          'bad network connection or difficulty reconciling multiple GraphDefs;' +
+          ' for the latter case, please refer to ' +
+          'https://github.com/tensorflow/tensorboard/issues/1929.');
+    })
+    .then((graph) => {
+      // Populate compatibile field of OpNode based on whitelist
+      tf.graph.op.checkOpsForCompatibility(graph, this.compatibilityProvider);
+
+      this._setOutGraph(graph);
+      var hierarchyTracker = tf.graph.util.getSubtaskTracker(tracker, 50,
+          'Namespace hierarchy');
+      return tf.graph.hierarchy.build(graph, hierarchyParams, hierarchyTracker);
+    })
+    .then((graphHierarchy) => {
+      // Update the properties which notify the parent with the
+      // graph hierarchy and whether the data has live stats or not.
+      this._setOutGraphHierarchy(graphHierarchy);
+    })
+    .catch((e) => {
+      // Generic error catch, for errors that happened outside
+      // asynchronous tasks.
+      const msg = `Graph visualization failed.\n\n${e}`;
+
+      tracker.reportError(msg, e);
+      // Don't swallow the error.
+      throw e;
+    });
+  },
+
+  _selectedFileChanged: function(e, overridingHierarchyParams) {
+    if (!e) {
+      return;
+    }
+    var file = e.target.files[0];
+    if (!file) {
+      return;
+    }
+
+    // Clear out the value of the file chooser. This ensures that if the user
+    // selects the same file, we'll re-read it.
+    e.target.value = '';
+
+    // Clear stats about the previous graph.
+    this._setOutStats(null);
```

Comment:
This line is new in this commit. Why?

Reply:
Must have mixed two commits by mistake. Will revert